File: cpp/sorting/selectionSort.cpp
# include <iostream>
using namespace std;

// BIG - O
// o(n^2)

void selectionSort(int arr[], int size) {
  int minValIndex;
  int temp;

  for (int i = 0; i < size; i++) {
    minValIndex = i;

    for (int j = i + 1; j < size; j++) {
      if (arr[j] < arr[minValIndex]) {
        minValIndex = j;
      }
    }

    temp = arr[i];
    arr[i] = arr[minValIndex];
    arr[minValIndex] = temp;
  }
}


int main() {
  int arr[]= {3, 5 ,22, 12, 43, 345, 65, 23};
  int size = sizeof(arr) / sizeof(arr[0]);
   selectionSort(arr, size);
  for (int i = 0; i < size; i++) {
    cout << arr[i] << "|";
  }
  return 0;
}

File: js/sorting/selectionSort.js
const selectionSort = function (arr) {
  let minVal;
  for (let i = 0; i < arr.length; i++) {
    minVal = i;
    for (let j = i; j < arr.length; j++) {
      if (arr[j] < arr[minVal]) {
        minVal = j;
      }
    }
    if (i !== minVal) [arr[i], arr[minVal]] = [arr[minVal], arr[i]];
  }
  return arr;
};

console.log(selectionSort([3, 4, 1, 2, 0, 1]));
